edge-modules/metrics-collector/src/Program.cs

```diff
@@ -50,12 +50,12 @@ static async Task<int> MainAsync()
             {
                 try
                 {
-                    moduleClientWrapper = await ModuleClientWrapper.BuildModuleClientWrapperAsync(Settings.Current.TransportType);
+                    moduleClientWrapper = await ModuleClientWrapper.BuildModuleClientWrapperAsync();
                     PeriodicTask periodicIothubConnect = new PeriodicTask(moduleClientWrapper.RecreateClientAsync, Settings.Current.IotHubConnectFrequency, TimeSpan.FromMinutes(1), LoggerUtil.Writer, "Reconnect to IoT Hub", true);
                 }
                 catch (Exception e)
                 {
-                    String msg = String.Format("Error connecting to Edge Hub. Is Edge Hub up and running with settings for clients connecting over {0}? Exception: {1}", Settings.Current.TransportType, e);
+                    String msg = String.Format("Error connecting to Edge Hub. This is only a fatal error if trying to upload to IoT Hub. For Azure Monitor upload, this connection is only used for Microsoft internal diagnostics. Exception: {0}", e);
```

Comment:
I'd rather see messages tailored to each case so the user doesn't have to do the mental mapping exercise. Also, we don't want to log the call stack in the `UploadTarget.AzureMonitor` case.

```csharp
                    if (Settings.Current.UploadTarget == UploadTarget.IotMessage)
                    {
                        LoggerUtil.Writer.LogError($"Error connecting to Edge Hub. Ensure Edge Hub is accepting client connections over {transportType}. Exception: {e}");
                        throw;
                    }
                    else
                    {
                        LoggerUtil.Writer.LogWarning($"Could not connect to Edge Hub to send usage data to Microsoft. This does not impact sending metrics to Azure Monitor. Diagnostic message: {e.Message}");
                    }
```

Reply:
I think we should actually plumb this catch logic deeper. Right now it is on the periodic task creation and initial connection. But if the initial connection succeeds then fails later customer will get logs about failing connection. They shouldn't in that case either.